src/Microsoft.AspNetCore.Mvc.Core/CompatibilityVersion.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// Specifies the version compatibility of settings configured by <see cref="MvcOptions"/>. 
+    /// </summary>
+    /// <remarks>
+    /// Setting <see cref="MvcCompatibilityOptions.CompatibilityVersion"/> to a specific version will 
+    /// change the default values of various settings to match a particular release of ASP.NET Core MVC. 
+    /// See <see cref="MvcCompatibilityOptions.CompatibilityVersion"/> for more details.
+    /// </remarks>
+    public enum CompatibilityVersion
+    {
+        /// <summary>
+        /// Sets the default value of settings on <see cref="MvcOptions"/> to match the behaviour of 
+        /// ASP.NET Core MVC 2.0.0.
```

Comment:
This will sound like a nit, but it isn't. The comments here specify the patch version, but the enum values do not. I think that neither should specify the patch version because that way it will be clearer that this is just about major/minor and not patches. We just won't change such behaviors in patches anyway - except when we do, but when we do, there will be quirks switches for it.

Reply:
Yeah I agree

Reply:
Would we consider using this mechanism for quirk switches? If we had a way to additionally pick up values for these switches from AppContext, it would make discovering \ using \ testing compat switches consistent both for users and us.

Reply:
There's no reason why we couldn't. I'm not sure if we'll find a case where that adds value though

Reply:
A requirement for quirks switches is that there has to be a way to enable them on .NET Core and .NET Framework *without recompiling* the app. So, if we can come up with something way to do that, it'd be an option.

Reply:
@Eilon I have to vet the design out, but the idea would be to read from `AppContext.GetData` as an additional source of values for these switches.

Reply:
`AppContext.TryGetSwitch()` is the way to read switches:

https://github.com/Eilon/QuirksModeSwitchSample/blob/master/QuirksModeSwitchSample/Program.cs#L27

But yeah, if we think that's a good feature to have here, that could be reasonable. We'll need to make sure every compat switch has a "reasonable" textual name to use in config files for quirks.

Reply:
I'm not convinced this is a good idea. I don't think these two things serve the same need.

Reply:
@rynowak I agree.